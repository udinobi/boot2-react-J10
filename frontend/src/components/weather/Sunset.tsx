import React from "react"

import styled from "styled-components"

import { SmallerText } from "../lib/StyledText"

export interface Data {
    sunset: string
}

const Sunset = styled(SmallerText)`
    display: inline;
    font-family: "Roboto", "sans-serif";
`

const Icon = styled.i`
    color: #00008B;
    font-size: 1.4rem;
    padding-right: 8px;
`

export default (props: Data) => 
    <div>
        <Icon className="wi wi-sunset" />
        <Sunset>{props.sunset}</Sunset>
    </div>
