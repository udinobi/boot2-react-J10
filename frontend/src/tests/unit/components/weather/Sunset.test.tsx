import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import Sunset, { Data } from "../../../../components/weather/Sunset"

describe("--> Snapshot for Components > weather > 'Sunset'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            sunset: ""
        }
    })

    it("matches", () => {
        const wrapper = shallow(<Sunset {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
