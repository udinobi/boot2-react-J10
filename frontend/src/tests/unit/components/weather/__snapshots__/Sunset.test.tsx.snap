// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'Sunset' matches 1`] = `
<div>
  <styled.i
    className="wi wi-sunset"
  />
  <Styled(Styled(Styled(styled.div))) />
</div>
`;
